.github/workflows/wheels.yml

```diff
@@ -130,16 +130,25 @@ jobs:
           CIBW_ENVIRONMENT_PASS_LINUX: RUNNER_OS
           CIBW_PRERELEASE_PYTHONS: True
 
+          # TODO remove the CIBW_BEFORE_BUILD_* CIBW_BEFORE_TEST lines once there are
+          #  numpy2.0 wheels available on PyPI.
+          # Also remove PIP_NO_BUILD_ISOLATION flag.
+          # For some reason asv needs extra packages for installation on cp312,
+          # but not on lower python versions
+          CIBW_BEFORE_BUILD_LINUX: "pip install numpy>=2.0.0.dev0 meson-python cython pythran pybind11 ninja; bash {project}/tools/wheels/cibw_before_build_linux.sh {project}"
+          CIBW_BEFORE_BUILD_WINDOWS: "pip install numpy>=2.0.0.dev0 meson-python cython pythran pybind11 ninja && bash {project}/tools/wheels/cibw_before_build_win.sh {project}"
+          CIBW_BEFORE_BUILD_MACOS: "pip install numpy>=2.0.0.dev0 meson-python cython pythran pybind11 ninja; bash {project}/tools/wheels/cibw_before_build_macos.sh {project}"
+          CIBW_BEFORE_TEST: pip install setuptools setuptools_scm[toml] wheel
```

Comment:
It looks like this is for installing `asv`. I filed an issue on the `asv` repo about missing wheels. However, we should not need `asv` to begin with in our wheel build jobs. It's not immediately clear to me why it's being pulled in.

Reply:
Asv is not needed in the build step, but during the test step. All wheels are tested during the cibuildwheel process

Reply:
Sure, but `asv` isn't a test dependency - it's for benchmarking, and we don't run benchmarks in the wheel build job.

Reply:
From looking at the git history it looks like I'm responsible (<chuckling>) for it being added to the [`test-requires` section](https://github.com/scipy/scipy/blob/main/pyproject.toml#L127). However, to be fair I was probably following it's presence in the entries in the [`project.optional-dependencies.test` section](https://github.com/scipy/scipy/blob/main/pyproject.toml#L80).

We can probably remove asv from the former, can we remove it from the latter?

Reply:
Yes. We can put it in a separate `benchmark` section instead I'd say, that'd be more accurate.